src/blocks/block-coauthor-image/edit.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Co-Author Feature Image
+ */
+
+import { __ } from '@wordpress/i18n';
+import {
+	useBlockProps,
+	store as blockEditorStore,
+	__experimentalUseBorderProps as useBorderProps,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import { TextControl, PanelBody, ToggleControl, SelectControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import DimensionControls from './dimension-controls';
+import PlaceholderImage from '../components/placeholder-image';
+import { getAvailableSizeSlug, getMediaSrc, getMediaDimensions, getPlaceholderImageDimensions } from './utils'
+
+/**
+ * Edit
+ *
+ * @export
+ * @param {Object} props { attributes, setAttributes, context, clientId }
+ * @return {WPElement}
+ */
+export default function Edit( { attributes, setAttributes, context, clientId } ) {
+
+	const { aspectRatio, height, isLink, rel, scale, sizeSlug, verticalAlign, width } = attributes;
+
+	// Author
+	const authorPlaceholder = useSelect(
+		select => select( 'co-authors-plus/blocks' ).getAuthorPlaceholder(),
+		[]
+	);
+	const author = context['co-authors-plus/author'] || authorPlaceholder;
+
+	// Media
+	const media = useSelect(
+		select => 0 !== author.featured_media && select( coreStore ).getMedia( author.featured_media, { context: 'view' } ),
+		[author.featured_media]
+	);
+
+	// Image Sizes and Dimensions
+	const { imageSizes, imageDimensions } = useSelect(
+		select => select( blockEditorStore ).getSettings(),
+		[]
+	);
+	const imageSizeOptions = imageSizes.map( ( { name, slug } ) => ({ value: slug, label: name }));
+	const availableSizeSlug = getAvailableSizeSlug( media, imageDimensions, sizeSlug );	
+	const dimensions = getMediaDimensions( media, imageDimensions, availableSizeSlug );
+	const placeholderDimensions = media ? {} : getPlaceholderImageDimensions(imageDimensions, availableSizeSlug);
+
+	// Border
+	const borderProps = useBorderProps( attributes );
+
+	// Don't placehold feature images for real authors with no image.
+	// Do placehold them in author archive contexts.
+	const panic = 0 !== author.id && false === media;
+
+	return (
+		<>
+			<DimensionControls
+				clientId={ clientId }
+				attributes={ attributes }
+				setAttributes={ setAttributes }
+				imageSizeOptions={ imageSizeOptions }
+			/>
+			{
+				panic ? null : (
+					<figure {...useBlockProps()}>
+						{
+							media ? (
+								<img
+									alt={__( 'Author featured image', 'co-authors-plus' )}
```

Comment:
Minor: I think you can remove the `co-authors-plus` text domain here, as you're specifying it in `block.json`:

https://github.com/Automattic/Co-Authors-Plus/blob/5638d641eb7ae8ed469a25c3c857f7a427e40fcf/src/blocks/block-coauthor-image/block.json#L63

I see you are already doing that other places, so let me know if that's intentional here. Thanks!

Reply:
Also seeing the same domain doubling in a some other `edit.js` block implementations.

Reply:
Oh, in a previous comment Gary asked that I explicitly include the text domain in the JS. I didn't know better one way or the other. Should I remove the text domain the JS as long as it's defined in the block.json?

Reply:
If Gary told you otherwise, then you're good to keep it as-is. Thanks!